pkg/asset/installconfig/azure/session.go

```diff
@@ -169,15 +201,13 @@ func checkCredentials(creds Credentials) error {
 	if creds.SubscriptionID == "" {
 		return errors.New("could not retrieve subscriptionId from auth file")
 	}
-	if creds.ClientID == "" {
-		return errors.New("could not retrieve clientId from auth file")
-	}
 	if creds.TenantID == "" {
 		return errors.New("could not retrieve tenantId from auth file")
 	}
-	if creds.ClientSecret == "" && creds.ClientCertificatePath == "" {
-		return errors.New("could not retrieve either client secret or client certs from auth file")
+	if (creds.ClientSecret != "" || creds.ClientCertificatePath != "") && creds.ClientID == "" {
```

Comment:
nit: technically you can have fewer checks if you check `creds.ClientID == ""` first

Reply:
`creds.ClientID` _can_ be `""` if using system Managed Identity. That's why I had to remove the check here https://github.com/openshift/installer/pull/7108/files#diff-da950a4ddabbede621d9d3b1058bb34f8931c89179306ee88a0e4d76a4cf0b13L172-L174. However, `ClientID` is still required for `ClientSecret` and `ClientCertificatePath`.